src/main/java/org/opensearch/sdk/ExtensionSettings.java

```diff
@@ -16,42 +16,37 @@
  */
 public class ExtensionSettings {
 
-    private String extensionname;
-    private String hostaddress;
-    private String hostport;
+    private String extensionName;
+    private String hostAddress;
+    private String hostPort;
 
     /**
      * Placeholder field. Change the location to extension.yml file of the extension.
      */
     public static final String EXTENSION_DESCRIPTOR = "src/test/resources/extension.yml";
 
-    public String getExtensionname() {
-        return extensionname;
-    }
-
-    public void setExtensionname(String extensionname) {
-        this.extensionname = extensionname;
-    }
-
-    public String getHostaddress() {
-        return hostaddress;
+    /**
+     * Jackson requires a default constructor.
+     */
+    private ExtensionSettings() {
```

Comment:
How does `ObjectMapper` use this?

Reply:
> How does `ObjectMapper` use this?

Reflection!

Reply:
It uses reflection to fill the attributes with the value present in the yml file

Reply:
Which introduces its own problems with Java Modules, if we ever decide to publish a module descriptor we'll have to open this package to Jackson's module.  But for now, let's enjoy the classpath...